metaseq/scripts/reshard_fsdp.py

```diff
@@ -135,7 +152,8 @@ def reshard_fsdp_model_weights(
             sharded_weights = []
             for weight, metadata in zip(shard_weights, shard_metadata):
                 pad = metadata["param_metadata"][idx]["params"][flat_name]["padding"]
-                sharded_weights.append(_unpad_tensor(weight[full_key], pad))
+                sharded_weight = _maybe_type(weight[full_key], dtype)
+                sharded_weights.append(_unpad_tensor(sharded_weight, pad))
```

Comment:
Any chance the order of unpad/type matters?

Reply:
I think such an order shouldn't matter here.